common/spritesheet.go

```diff
@@ -4,23 +4,58 @@ import (
 	"log"
 
 	"engo.io/engo"
+	"engo.io/engo/math"
 	"engo.io/gl"
 )
 
 // Spritesheet is a class that stores a set of tiles from a file, used by tilemaps and animations
 type Spritesheet struct {
-	texture               *gl.Texture     // The original texture
-	width, height         float32         // The dimensions of the total texture
-	cellWidth, cellHeight int             // The dimensions of the cells
-	cache                 map[int]Texture // The cell cache cells
+	texture       *gl.Texture     // The original texture
+	width, height float32         // The dimensions of the total texture
+	cells         []Sprite        // The dimensions of each sprite
+	cache         map[int]Texture // The cell cache cells
 }
 
-func NewSpritesheetFromTexture(tr *TextureResource, cellWidth, cellHeight int) *Spritesheet {
-	return &Spritesheet{texture: tr.Texture,
-		width: tr.Width, height: tr.Height,
-		cellWidth: cellWidth, cellHeight: cellHeight,
-		cache: make(map[int]Texture),
+// Sprite holds the position data for each sprite on the sheet
+type Sprite struct {
```

Comment:
Maybe it'd be better to call this a 'TextureLocation' or 'SpriteLocation' (or maybe 'XRegion')

Reply:
Updated to SpriteRegion. Thanks, that is much more clear ^_^